Fix pagination documentation for stellar-rpc

tamirms opened the pull request:
Fixes https://github.com/stellar/stellar-docs/issues/1309

This PR consolidates the documentation for pagination that is used for getEvents, getTransactions, and getLedgers. All those endpoints use a `cursor` field in the request / response in order to do pagination.

The `cursor` field used to be called `paging_token` and this PR updates the documentation to reference `cursor` instead of `paging_token`.

Also, the `cursor` field was missing in the schema of the `getEvents` and `getTransactions` responses. This PR includes the `cursor` fields in those responses.

stellar-jenkins commented:
**Something went wrong with PR preview build please check**

stellar-jenkins commented:
**Something went wrong with PR preview build please check**

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr1363.previews.kube001.services.stellar-ops.com

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr1363.previews.kube001.services.stellar-ops.com

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr1363.previews.kube001.services.stellar-ops.com

Shaptic commented on openrpc/src/stellar-rpc/methods/getEvents.json:
```diff
@@ -39,7 +39,7 @@
                 "cursor": {
                     "type": "string",
-                    "$ref": "#/components/schemas/GetEventsCursor"
+                    "$ref": "#/components/schemas/Cursor"
```
Each event itself also has a cursor - I think that needs to be updated, too?

tamirms replied:
in the event schema the field is still called `pagingToken`:

https://github.com/stellar/stellar-rpc/blob/main/protocol/get_events.go#L29-L30

Shaptic replied:
Oh oof, gotcha. That one is still useful for mid-page eventing, though, so idk if deprecated is the word? Then again they should use `id` for that.

Shaptic reviewed (commented):
Nice! You need to regenerate the "compiled" RPC spec, though:

```
yarn rpcspec:build
```

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr1363.previews.kube001.services.stellar-ops.com

tamirms commented:
@Shaptic can you take another look? I updated the pr with the following changes:

- ran `yarn rpcspec:build`
- added a note to the `pagingToken` field within the event schema that it's deprecated in favor of the top level cursor
- fixed the link to the pagination page (previously it was hardcoded to the developers.stellar.org domain)
